### PR TITLE
Support allureId attribute to permanent mapping result to existing test case

### LINKED ISSUE #25: Support allureId attribute to permanent mapping result to existing test case
**Current state and problem**
When allure_result/*.json are uploaded to Allure TestOps, it performs automatic mapping between test case and result, based on `<package>.<class name>.<method name>`. When test launch is closed, automatic test cases are created, and all next results are linked to it. 

It's recommended to specify the id of test case to avoid wrong automapping after test code changes, using allureId in terminology, don't confuse with Allure.Commons "testCaseId" property, which is id from external TMS.

More details at https://docs.qameta.io/allure-testops/ecosystem/intellij-plugin/#link-tests-in-ide-to-tests-cases-in-allure-testops-via-allureid

**Proposed solution**
 [AllureAllureId("1")] attribute is created, internally it adds label with name "as_id", same as java version do. 

https://javadoc.io/static/io.qameta.allure/allure-java-commons/2.13.9/io/qameta/allure/AllureId.html https://github.com/allure-framework/allure-java/blob/708a1f113842fadfc6d0cb092bb084b319525a49/allure-java-commons/src/main/java/io/qameta/allure/util/ResultsUtils.java#L81

**Testing**
Before fix: get result*.json, upload to Allure TestOps test launch and finish it, it creates automatic test case. Upload another test run results again, it's automapped to same test case. Try to change test code namespace for example, upload results, then new automaped test case is created, no connection to a previous one.

After fix:
get result*.json, upload to Allure TestOps, it will always be mapped to specified test case id.

Unit tests:
One extra tests is created, in result*.json we have label with name="as_id" and value.

image

### PR DESCRIPTION
**Pull request by sphinxy**
Fixes #25

### CODE REVIEW

File: src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Allure.Xunit.Attributes
+{
+    /// <summary>
+    /// Internal AllureId of test case, used to permanent mapping result to existing automatic test case. NB: don't confuse with "test case id" aka id for external TMS system.
+    /// https://docs.qameta.io/allure-testops/ecosystem/intellij-plugin/#link-tests-in-ide-to-tests-cases-in-allure-testops-via-allureid
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AllureAllureIdAttribute : Attribute, IAllureInfo
```
Comment (rus-art) [inline src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs]: Can we use name `AllureIdAttribute`? `AllureAllureId` is slightly strange

Comment (sphinxy) [reply src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs]: Yes, a little bit strange from the first glance, but it's how it should be named to stay consistent. 

AllureID is the actual name from Allure docs, e.g. "Allure TestOps also uses the parameters of a test to link a test case to certain AllureID, i.e. test <package>.<class name>.<method name> without parameters and with one parameter aren’t the same, these will be two different test cases with different AllureIDs." and the prefix Allure is used from all attributes, so
AllureEpic
AllureLabel
AllureAllureId
AllureXunitTheory


Just using AllureID with means "prefix Allure plus ID", but ID is not exists in allure terminology, on AllureID exists => AllureAllureId as "prefix Allure plus AllureId parameter"

Comment (rus-art) [reply src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs]: @neparij Do you agree with such naming?

Comment (sphinxy) [reply src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs]: Personally I would accept any naming, just please choose anything that fits and accept PR

Comment (AzatKalim) [reply src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs]: Hi @sphinxy, If you change class name AllureAllureIdAttribute to AllureIdAttribute we will accept this PR

Comment (sphinxy) [reply src/Allure.XUnit/Attributes/AllureAllureIdAttribute.cs]: @AzatKalim fixed.

**Review by AzatKalim — APPROVED** [review]

**Review by rus-art — APPROVED** [review]

**Pull request merged**